src/journal/journald-kmsg.c

```diff
@@ -397,11 +397,17 @@ static int dispatch_dev_kmsg(sd_event_source *es, int fd, uint32_t revents, void
 }
 
 int server_open_dev_kmsg(Server *s) {
+        mode_t mode;
         int r;
 
         assert(s);
 
-        s->dev_kmsg_fd = open("/dev/kmsg", O_RDWR|O_CLOEXEC|O_NONBLOCK|O_NOCTTY);
+        if (s->read_kmsg)
+                mode = O_RDWR|O_CLOEXEC|O_NONBLOCK|O_NOCTTY;
+        else
+                mode = O_WRONLY|O_CLOEXEC|O_NONBLOCK|O_NOCTTY;
```

Comment:
The condition `_unlikely_(s->dev_kmsg_fd < 0)` is still used to decide whether `journald` have to forward messages to `kmsg`. Do you think that makes sense when `true` is assigned to `ReadKMsg`? Could you tell how exactly the patch has been tested?